File: server/src/main/java/edu/cnm/deepdive/coffeeshop/service/InterestService.java
package edu.cnm.deepdive.coffeeshop.service;

import edu.cnm.deepdive.coffeeshop.model.dto.InterestCreateDto;
import edu.cnm.deepdive.coffeeshop.model.dto.InterestDto;
import java.util.List;
import java.util.UUID;

public interface InterestService {

  InterestDto getInterest(UUID interestId);

  List<InterestDto> getAllInterests();

  InterestDto createInterest(InterestCreateDto interest);
}

File: app/src/main/java/edu/cnm/deepdive/coffeeshop/service/InterestService.kt
package edu.cnm.deepdive.coffeeshop.service

import edu.cnm.deepdive.coffeeshop.model.domain.Interest
import edu.cnm.deepdive.coffeeshop.repository.InterestRepository
import jakarta.inject.Inject
import jakarta.inject.Singleton
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.SupervisorJob
import kotlinx.coroutines.future.future
import java.util.UUID
import java.util.concurrent.CompletableFuture

@Singleton
class InterestService @Inject constructor(private val interestRepository: InterestRepository){

    private val scope = CoroutineScope(SupervisorJob() + Dispatchers.IO)

    fun getInterests(): CompletableFuture<List<Interest>> =
        scope.future {
        interestRepository.getInterests()
    }

    fun getInterest(id: UUID): CompletableFuture<Interest> =
        scope.future {
        interestRepository.getInterest(id)
    }
}